crud/select.lua

```diff
@@ -1,35 +1,40 @@
-local checks = require('checks')
 local errors = require('errors')
-local vshard = require('vshard')
-local fun = require('fun')
 
-local call = require('crud.common.call')
-local utils = require('crud.common.utils')
-local sharding = require('crud.common.sharding')
-local dev_checks = require('crud.common.dev_checks')
-
-local select_conditions = require('crud.select.conditions')
-local select_plan = require('crud.select.plan')
 local select_executor = require('crud.select.executor')
-local select_comparators = require('crud.select.comparators')
 local select_filters = require('crud.select.filters')
-
-local Iterator = require('crud.select.iterator')
+local dev_checks = require('crud.common.dev_checks')
 
 local SelectError = errors.new_class('SelectError')
-local GetReplicasetsError = errors.new_class('GetReplicasetsError')
 
-local select_module = {}
+local select_module
+
+-- "merger" segfaults here
+-- See https://github.com/tarantool/tarantool/issues/4954
+if '2' <= _TARANTOOL and _TARANTOOL <= '2.3.3' then
+    select_module = require('crud.select.compat.select_old')
+else
+    -- Try to require built-in or external merger
+    local ok, module = pcall(require, 'crud.select.compat.select')
+    if ok then
+        select_module = module
+    else
+        select_module = require('crud.select.compat.select_old')
```

Comment:
Are you sure that we shouldn't fail here?

Reply:
Hope Lua-implemented merger won't fail

Reply:
NB:

- crud.select.compat.select - default module with merged
- crud.select.compat.select_old - compatibility module without merger support written in Lua

@AnaNek maybe it would be great to reflect it in commit message